configs/example/gem5_library/multisim/multisim-fs-x86-npb.py

```diff
@@ -0,0 +1,140 @@
+
+```shell
+<gem5-binary> -m gem5.utils.multisim \
+    configs/example/gem5_library/multisim/multisim-fs-x86-npb.py
+```
+
+2. To run a specific simulation defined in this script:
+
+```shell
+<gem5-binary> configs/example/gem5_library/multisim/multisim-fs-x86-npb.py \
+    <process_id> # e.g. npb-bt-a_cores-1
+```
+
+3. To list all the IDs of the simulations defined in this script:
+
+```shell
+<gem5-binary> configs/example/gem5_library/multisim/multisim-fs-x86-npb.py -l
+```
+"""
+
+import m5
+
+import gem5.utils.multisim as multisim
+from gem5.coherence_protocol import CoherenceProtocol
+from gem5.components.boards.x86_board import X86Board
+from gem5.components.memory import DualChannelDDR4_2400
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.components.processors.simple_switchable_processor import (
+    SimpleSwitchableProcessor,
+)
+from gem5.isas import ISA
+from gem5.resources.resource import obtain_resource
+from gem5.simulate.simulator import (
+    ExitEvent,
+    Simulator,
+)
+from gem5.utils.requires import requires
+
+requires(
+    isa_required=ISA.X86,
+    coherence_protocol_required=CoherenceProtocol.MESI_TWO_LEVEL,
+)
+
+from gem5.components.cachehierarchies.ruby.mesi_two_level_cache_hierarchy import (
+    MESITwoLevelCacheHierarchy,
+)
+
+
+def handle_workbegin():
+    m5.stats.reset()
+    processor.switch()
+    yield False
+
+
+def handle_workend():
+    m5.stats.dump()
+    yield True
+
+
+# Here we imagine an examperiment wanting to run each NPB benchmark on the same
+# system twice: once with 1 core and once with 2 cores.
+for benchmark in obtain_resource("npb-benchmark-suite"):
+    for num_cores in [1, 2]:
+        # Limit the number of simulators, so we don't run a massive amount of
+        # them in the proof-of-concept.
+        if multisim.num_simulators() >= 6:
+            break
+
+        cache_hierarchy = MESITwoLevelCacheHierarchy(
+            l1d_size="32kB",
+            l1i_size="32kB",
+            l2_size="256kB",
+            l1d_assoc=8,
+            l1i_assoc=8,
+            l2_assoc=16,
+            num_l2_banks=2,
+        )
+        memory = DualChannelDDR4_2400(size="3GB")
+        processor = SimpleSwitchableProcessor(
+            starting_core_type=CPUTypes.ATOMIC,
+            switch_core_type=CPUTypes.TIMING,
+            isa=ISA.X86,
+            num_cores=num_cores,
+        )
+        board = X86Board(
+            clk_freq="3GHz",
+            processor=processor,
+            memory=memory,
+            cache_hierarchy=cache_hierarchy,
+        )
+
+        board.set_workload(benchmark)
+
+        simulator = Simulator(
+            board=board,
+            on_exit_event={
+                ExitEvent.WORKBEGIN: handle_workbegin(),
+                ExitEvent.WORKEND: handle_workend(),
+            },
+        )
+
+        simulator.set_id(f"{benchmark.get_id()}_cores-{num_cores}")
+
+        multisim.add_simulator(simulator)
```

Comment:
I think it would be better to explicitly run the simulations. E.g.,

```suggestion
        multisim.add_simulator(simulator)
        
multisim.run_all(num_processes=4)
```

Reply:
Ok, I have thought about it a bit more, and I'm not sure if I agree with myself... We need *something* here, but I don't know what. Maybe something like

```suggestion
        if __name__ == "__m5_main__" or if __name__ == "__main__": 
            print(f"""This script is meant to be run with the `multisim` module
You can run it with `<gem5 binary> -m gem5.multiprocessing.multisim -p <processes> {__file__}
See ??? for more information"""
           
```

Reply:
I'm not really sure what's actionable here. There's already code which gives this error in the multisim module: that is if a user uses `multisim.add_simulator` outsi@de of the `gem5.multiprocessing.multisim` module, so i don't see the use in adding this to the configs.